### PR TITLE
Java9 fixes

### PR DESCRIPTION
**Pull request by peter-lawrey**
This two-patch set brings **better test coverage** for `AffinityLock` edge-cases **and full Java 9+ compatibility** across the project.

---

### 1️⃣  New unit-tests

* **`AffinityLockTest`**
  * Verifies non-binding acquisition leaves the calling thread’s mask unchanged.
  * Confirms illegal CPU IDs and double-bind scenarios raise/behave as expected.
* **`FileLockLockCheckTest`**
  * Covers lock-file deletion while held and requests for CPUs with no lock file.
* **`LockCheckTest`**
  * Adds negative-PID sanity check.
  
_Total: +49 LOC in `AffinityLockTest` and +30 +  lines in lock-checker tests._

---

### 2️⃣  Java 9+ support & misc fixes

| Area | Update |
|------|--------|
| **Maven native profile** | `make-c` now activates only on **Linux x86_64** (skips ARM and non-Linux builds). |
| **Boot class-path scanner** | `BootClassPath` falls back to walking the modular **`jrt:/`** filesystem when `sun.boot.class.path` is absent (JDK 9+). |
| **Process ID retrieval** | New util `Utilities.currentProcessId()` uses `ProcessHandle.current().pid()` when available; all PID calls routed through it (`LockCheck`, `NullAffinity`, macOS/Solaris impls, etc.). |
| **LockCheck** | Guards `getProcessForCpu()` on unsupported OS, uses new PID helper. |
| **WindowsJNAAffinity** | Safer “set affinity” verification and SLF4J-parameterised logging. |
| **Generic logging / warnings** | Switched to parameterised SLF4J in several files. |
| **BootClassPathTest** | No longer restricted to JDK 8. |

---

### Behavioural notes

* Invalid CPU-ID acquisition now returns an *unbound* lock instead of throwing, so assertions were adjusted in tests.
* Lock-file logic robust to manual deletion on disk.
* Java 11+ builds succeed without additional flags; class-path scanning works inside modules.

> These changes strengthen edge-case validation and ensure the library runs cleanly on modern JDK versions while keeping legacy behaviour intact.

### CODE REVIEW

File: affinity/src/test/java/net/openhft/affinity/AffinityLockTest.java [hunk 1]
```diff
@@ -322,16 +323,55 @@ public void testAffinityLockDescriptions() {
...
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException ignored) {
+                // ignored
+            }
+        });
+        t.start();
+
+        while (!lock.isBound()) {
+            Thread.sleep(10);
```
Comment (tgd) [inline affinity/src/test/java/net/openhft/affinity/AffinityLockTest.java]: Consider replacing with `Waiters.await...` with an upper time bound to stop this hanging in CI if an error occurs

File: affinity/src/main/java/net/openhft/affinity/impl/Utilities.java [hunk 2]
```diff
@@ -79,4 +79,30 @@ private static boolean is64Bit0() {
...
+     * Returns the current process id. Uses {@code ProcessHandle} when running
+     * on Java&nbsp;9 or later and falls back to parsing
+     * {@code RuntimeMXBean#getName()} on earlier versions.
+     *
+     * @return the process id or {@code -1} if it cannot be determined
+     */
+    public static int currentProcessId() {
+        try {
+            // Java 9+ provides ProcessHandle which has a pid() method.
+            Class<?> phClass = Class.forName("java.lang.ProcessHandle");
```
Comment (tgd) [inline affinity/src/main/java/net/openhft/affinity/impl/Utilities.java:93]: Consider PosixAPI#getpid

https://github.com/OpenHFT/Posix/blob/d73e85611336a7d43111f2471cc010cf4b81f6c7/src/main/java/net/openhft/posix/PosixAPI.java#L485

File: affinity/src/test/java/net/openhft/affinity/AffinityLockTest.java (hunk 1 above)
Comment (peter-lawrey) [reply affinity/src/test/java/net/openhft/affinity/AffinityLockTest.java]: Good idea

File: affinity/src/main/java/net/openhft/affinity/impl/Utilities.java (hunk 2 above)
Comment (peter-lawrey) [reply affinity/src/main/java/net/openhft/affinity/impl/Utilities.java:93]: Thread Affinity should probably use the PosixAPI but it doesn't.
